### PR TITLE
Use Jest to run tests

### PR DESCRIPTION
**Pull request by wylieconlon**
This PR migrates all the tests to run using Jest, which also involved removing the [`esm`](https://www.npmjs.com/package/esm) loader and using the updated support for modules in Node 14. The rollup output is unchanged.

Migrating the tests to Jest was necessary so that I can upgrade the Tinymath syntax for Lens usage.

### CODE REVIEW

File: rollup.bundles.js [hunk 1]
```diff
@@ -21,42 +21,7 @@ const config = ({ minify = false } = {}) => {
     include: 'src/**',
   };
 
-  const babelConfig = {
-    exclude: 'node_modules/**', // only transpile our source code
-    presets: [
-      [
-        '@babel/preset-env',
-        {
-          modules: false,
```
Comment (wylieconlon) [inline rollup.bundles.js:30]: This line is the only one I didn't copy into babel.config.cjs

**Comment by spalger** [comment]
I think we should close this and archive the repo, thoughts?

**Comment by wylieconlon** [comment]
@spalger That seems reasonable to me, but let's wait to merge https://github.com/elastic/kibana/pull/89383 first

**Comment by wylieconlon** [comment]
Closing because it's merged into Kibana

**Pull request closed without merging**